public/js/icinga/behavior/navigation.js

```diff
@@ -282,63 +307,101 @@
     };
 
     /**
-     * Show the fly-out menu
+     * Captures the mouse enter events to the navigation item and show the flyout.
      *
      * @param e
      */
-    Navigation.prototype.showFlyoutMenu = function(e) {
-        var $layout = $('#layout');
-
+    Navigation.prototype.onMouseEnter = function(e) {
+        const $layout = $('#layout');
+        const _this = e.data.self;
         if ($layout.hasClass('minimal-layout')) {
             return;
         }
 
-        var $target = $(this);
-        var $flyout = $target.find('.nav-level-2');
+        const $target = $(this);
+
+        if (
+            d3 !== null
+            && ! _this.extendedFlyoutZone.includes(undefined)
+            && d3.polygonContains(_this.extendedFlyoutZone, [e.clientX, e.clientY])
+        ) {
+            return;
+        }
 
-        if (! $flyout.length) {
+        if (! $target[0].matches(':has(.nav-level-2)')) {
             $layout.removeClass('menu-hovered');
             $target.siblings().not($target).removeClass('hover');
             return;
         }
 
-        var delay = 300;
+        if (! $target.is(':hover')) {
+            return;
+        }
 
-        if ($layout.hasClass('menu-hovered')) {
-            delay = 0;
+        $layout.addClass('menu-hovered');
+        _this.extendedFlyoutZone[0] = [e.clientX, e.clientY];
+        _this.showFlyoutMenu($target);
+    }
+
+    /**
+     * Captures the mouse move events within the navigation item
+     * and show the flyout if needed.
+     *
+     * @param e
+     */
+    Navigation.prototype.onMouseMove = function(e) {
+        const _this = e.data.self;
+        clearTimeout(_this.flyoutTimer);
+
+        const $target = $(this);
+
+        if (
+            $target[0].matches(':has(.nav-level-2)')
+            && ! $target.hasClass('hover')
+            && ! _this.extendedFlyoutZone.includes(undefined)
+            && d3.polygonContains(_this.extendedFlyoutZone, [e.clientX, e.clientY])
```

Comment:
Something is off here. For me, the flyout doesn't disappear after 200ms even though I'm clearly not over the original target anymore. I'll show in person how this behaves.